### ISSUE TITLE
How should I train objectron framework with my own dataset?

### ISSUE THREAD

**Issue by dbxinj** [google/mediapipe#527]
Hi,

I've tried your 3d object detection and tracking demo on android. The results look quite promising.

However, I also noticed that the training part of 3d object detection model (https://arxiv.org/pdf/2003.03522.pdf) is not included in the repo. 

Actually, I would like to train the 3d object detection model to my own dataset, which may include some other objects, like sunglasses. I'm interested about whether you are going to release the training code so that it would be easier to adapt it to other user cases, instead of re-implementing based on the paper?

**Comment by mgyong** [comment]
@dbxinj We do not plan to release the training pipeline but there is potentially a plan to release the dataset and also for external contributions to the dataset

**Comment by dbxinj** [comment]
Understood, but may I ask for more details about the 3d object detection model as described in the paper? Not sure if it's appropriate to ask here, but I just post it first. 

Firstly, what's the exact input labels fed to the network during training? Can you give me one example, i.e., one image with corresponding label file?

Secondly, about detection head, I still do not fully understand how you get the prediction and ground-truth heatmap. 
For prediction heatmap, do you do max-pooling or avg-pool along channel axis, or maybe just a convolution to compress 40x30x?? to 40x30x1 heatmap? 
For ground truth map, you generate the heatmap by bivariate normal distribution? Since the ground truth labels only include the center points and box coordinates for each object, you can get the object height and width in 2D. Then get the distribution with these fixed parameters for each object.

Thirdly, about regression head, I still have some confusions about the equation(3). Predicted Di(p) are the values from the 40x30x2 feature map. The predicted values from the network are the offsets only, right? Then, Di(p) + p gets the full predicted box coordinates, while xi is the ground-truth coordinates.
Are you doing regression within a certain range of area (h > 0.2) for each corner vertex of the box, instead of just for the exact 8 vertices?

**Comment by alter-sachin** [comment]
Ping.
